### PR TITLE
Optionally use the AnitomySharp library to extract media titles

### PR DESCRIPTION
**Pull request by MisterMcDuck**
I'm not sure if there will be much interest in this PR considering that it seems like many people using Jellyfin organize their media to use tvdb naming schemes, but this change adds the option to extract the titles of media folders for Anilist searches. 
e.g. 
[Golumpa] A Certain Scientific Railgun T (Toaru Kagaku no Railgun T) [FuniDub 1080p x264 AAC]
becomes
A Certain Scientific Railgun T

for the initial Anilist search. If this search fails, it falls back on the existing logic of exact folder searches. 

This allowed me to quickly import my large unmodified media collection with reasonable success (~85% hit rate), rather than renaming all of the folders to match the Anilist/tvdb names. 

Note: Due to the .NET jellyfin version I used a fork of AnitomySharp from @nielsvanvelzen located at https://github.com/nielsvanvelzen/AnitomySharp
This would be an external dependency, which I'm not sure is allowed/encouraged.

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
I can probably look into publishing the library to NuGet if required. Might take some time though.

**Comment by Xabis** [comment]
Seems the AnitomySharp project is out of maintenance, As such, using the nuget package is not recommended.

I forked and updated the library to match the latest changes from erengy here: https://github.com/Xabis/AnitomySharp

----

**Maintainers**: is there something blocking this PR from moving forward? This has been open for 2 years now. I made similar modifications to this plugin locally, before stumbling on to this submission, so there is certainly interest here.

**Comment by cuanim** [comment]
> Maintainers: is there something blocking this PR from moving forward? This has been open for 2 years now. I made similar modifications to this plugin locally, before stumbling on to this submission, so there is certainly interest here.


+1

**Comment by crobibero** [comment]
One thing that is blocking this PR is that it doesn't build. If someone would like to recreate this PR with the new AnitomySharp library feel free

**Comment by MisterMcDuck** [comment]
I've added a reference to a net6 compatible AnitomySharp nuget package which I found. It compiles locally for me.

File: Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs
Comment (crobibero) [inline Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs:19]:
```suggestion
                @"(\s|\.)S[0-9]{1,2}", String.Empty);
```

File: Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs
Comment (crobibero) [inline Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs:23]:
```suggestion
                @"\s*~(\w|[0-9]|\s)+~", String.Empty);
```

File: Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs
Comment (crobibero) [inline Jellyfin.Plugin.AniList/Providers/AniList/AniListSearchHelper.cs:29]:
```suggestion
                @"\((\w|[0-9]|\s)+\)$", String.Empty);
```

**Review by crobibero — APPROVED** [review]

**Pull request merged**
